pkg/imgpkg/bundle/locations_config.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package bundle
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	LocationFilepath   = "images-locations.yml"
+	LocationKind       = "Locations"
+	LocationAPIVersion = "imgpkg.carvel.dev/v1alpha1"
+)
+
+type LocationConfig struct {
+	APIVersion string          `json:"apiVersion"` // This generated yaml, but due to lib we need to use `json`
+	Kind       string          `json:"kind"`       // This generated yaml, but due to lib we need to use `json`
+	Images     []ImageLocation `json:"images"`     // This generated yaml, but due to lib we need to use `json`
+}
+
+type ImageLocation struct {
+	Image string `json:"image"` // This generated yaml, but due to lib we need to use `json`
+	// Repository will not be used right now because all images will be in the bundle repository
+	Repository string `json:"repository,omitempty"` // This generated yaml, but due to lib we need to use `json`
+	Bundle     bool   `json:"bundle"`               // This generated yaml, but due to lib we need to use `json`
+}
+
+func NewLocationConfigFromPath(path string) (LocationConfig, error) {
+	bs, err := ioutil.ReadFile(path)
+	if err != nil {
+		return LocationConfig{}, fmt.Errorf("Reading path %s: %s", path, err)
+	}
+
+	return NewLocationConfigFromBytes(bs)
+}
+
+func NewLocationConfigFromBytes(data []byte) (LocationConfig, error) {
+	var lock LocationConfig
+
+	err := yaml.UnmarshalStrict(data, &lock)
```

Comment:
hmm, whenever we add new fields, is "strict" going to screw us up?

Reply:
This function does not allow unknown fields. This will only cause problems if you are using an older version of `imgpkg` with a newer version of the Image Locations configuration. To be fair I think we want that because when we reintroduce the Repository field we need to respect it.